src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -95,4 +95,8 @@ public class QuicStreamAbortedException : QuicException
         public QuicStreamAbortedException(string message, long errorCode) : base(message) { }
         public long ErrorCode { get; }
     }
+    public class QuicOperationAbortedException : QuicException
```

Comment:
I see that QuicException derives directly from Exception. Would it be better to have it derive from IOException or possibly OperationCanceledException if it's always user-initiated?

I would lean towards IOException given how it's currently used. I see it's being triggered by stuff like "HandleEventShutdownInitiatedByPeer" which feels really IOExceptiony to me.

Reply:
Yea that seems reasonable. Open up an issue for it?

Reply:
I also don't know if QuicOperationAbortedException will be necessary now if we start returning null.

Reply:
I looked at SocketException, and it doesn't derive from IOException, so I'm no longer convinced this is the way to go. Maybe it should derive from SystemException (which SocketException does indirectly), but I'm not sure what the guidance there is. Is this supposed to be the base class for all Exceptions in a System.* namespace?